private/buf/cmd/buf/command/generate/generate.go

```diff
@@ -524,67 +480,42 @@ func getInputImages(
 	return inputImages, nil
 }
 
-// getBufGenYAMLFileWithFlagEquivalence returns a buf gen yaml file the same as
-// the given one, except that it is overriden by flags.
-// This is called only for migration. Input will be used regardless if it's empty.
-func getBufGenYAMLFileWithFlagEquivalence(
-	ctx context.Context,
-	logger *zap.Logger,
-	bufGenYAMLFile bufconfig.BufGenYAMLFile,
-	input string,
-	flags flags,
-) (bufconfig.BufGenYAMLFile, error) {
-	if input == "" {
-		input = "."
-	}
-	inputConfig, err := buffetch.GetInputConfigForString(
-		ctx,
-		buffetch.NewRefParser(logger),
-		input,
-	)
-	if err != nil {
-		return nil, err
-	}
-	var includeTypes []string
-	if bufGenYAMLFile.GenerateConfig().GenerateTypeConfig() != nil {
-		includeTypes = bufGenYAMLFile.GenerateConfig().GenerateTypeConfig().IncludeTypes()
-	}
-	if len(flags.Types) > 0 {
-		includeTypes = flags.Types
-	}
-	inputConfig, err = bufconfig.NewInputConfigWithTargets(
-		inputConfig,
-		flags.Paths,
-		flags.ExcludePaths,
-		includeTypes,
-	)
-	if err != nil {
-		return nil, err
-	}
-	pluginConfigs, err := slicesext.MapError(
-		bufGenYAMLFile.GenerateConfig().GeneratePluginConfigs(),
-		func(pluginConfig bufconfig.GeneratePluginConfig) (bufconfig.GeneratePluginConfig, error) {
-			return bufconfig.NewGeneratePluginWithIncludeImportsAndWKT(
-				pluginConfig,
-				flags.IncludeImports,
-				flags.IncludeWKT,
-			)
+// TODO: where does this belong? A flagsext package?
+// value must not be nil.
+func bindBoolPointer(flagSet *pflag.FlagSet, name string, value **bool, usage string) {
+	flag := flagSet.VarPF(
+		&boolPointerValue{
+			valuePointer: value,
 		},
+		name,
+		"",
+		usage,
 	)
-	if err != nil {
-		return nil, err
+	flag.NoOptDefVal = "true"
+}
+
+// Implements pflag.Value.
+type boolPointerValue struct {
+	// This must not be nil at construction time.
```

Comment:
This exists because I couldn't find an out-of-the-box function in `pflags` library (something like `BindBoolVarPointer`).